src/pages/auth/hooks/useEmailPrompt.ts

```diff
@@ -0,0 +1,77 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { useAuthContext } from '../../../context'
+import { useUserEmailUpdateMutation } from '../../../types'
+import { authUserAtom } from '../state'
+
+type FormValues = Record<string, any>
+
+const emailSchema = yup.object().shape({
+  dontAskAgain: yup.boolean(),
+  email: yup
+    .string()
+    .email('Invalid email')
+    .when('dontAskAgain', {
+      is: false,
+      then: (schema) => schema.required('Email is required'),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+})
+
+export const dontAskAgainAtom = atomWithStorage('dontAskAgain', false)
+
+export const shouldPromptAtom = atom((get) => {
+  const user = get(authUserAtom)
+  const dontAskAgain = get(dontAskAgainAtom)
+  return user.id && !user.email && !dontAskAgain
+})
+
+export const useEmailPrompt = () => {
+  const { user, setUser } = useAuthContext()
+  const [enableSave, setEnableSave] = useState(false)
+  const setDontAskAgain = useSetAtom(dontAskAgainAtom)
+  const shouldPrompt = useAtomValue(shouldPromptAtom)
+  const [updateUserEmail] = useUserEmailUpdateMutation({
+    onCompleted(data) {
+      if (data?.userEmailUpdate) {
+        setUser({ ...user, email: data.userEmailUpdate.email })
+        setDontAskAgain(true)
+      }
+    },
+  })
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors, isValid, isDirty },
+    reset,
+  } = useForm<FormValues>({
+    resolver: yupResolver(emailSchema),
+    defaultValues: {
+      email: user.email,
+      dontAskAgain: false,
+    },
+  })
+
+  useEffect(() => {
```

Comment:
you don't need a state and useEffect for this.
I think the formState values, when updated they already are reactive, so they should automatically rerender, without having to need an extra state. 
 a simple, 
 
 `const enableSave = isValid && isDirty `
 
 should do the same thing.